### PR TITLE
Automated cherry pick of #76656: Switch to instance-level update APIs for Azure VMSS loadbalancer operations

### PR DESCRIPTION
**Pull request by feiskyer**
Cherry pick of #76656 on release-1.12.

#76656: Switch to instance-level update APIs for Azure VMSS loadbalancer operations

Previously, VMSS API doesn't support configurations for each instance. Hence, Azure cloud provider upgrades manually for all instances whenever loadbalancer backend pools need to be updated. This would cause all instances restarted if there're also some other VMSS model changes.

To fix this issue, the PR switches to instance-level update APIs for Azure VMSS loadbalancer operations. It also upgrades Azure SDK to v21.4.0.

/kind bug
/priority critical-urgent
/sig azure

### CODE REVIEW

**Comment by justaugustus** [comment]
For SIG Azure & Release:
/lgtm
/approve

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Comment by cblecker** [comment]
verify failure is legitimate. please review and fix.

**Comment by feiskyer** [comment]
@cblecker gofmt fixed now, I'm using a wrong Go version initially. PTAL

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"ImportPath": "k8s.io/kubernetes",
-	"GoVersion": "go1.10",
+	"GoVersion": "go1.12",
```
Comment (cblecker) [inline Godeps/Godeps.json]: We shouldn't be changing the go version here. 1.12 is still on go1.10

Comment (feiskyer) [reply Godeps/Godeps.json]: good catch. Let me update the vendor again by the correct Go version

Comment (feiskyer) [reply Godeps/Godeps.json]: fixed now

**Comment by justaugustus** [comment]
Commented on the flake issue opened here: https://github.com/kubernetes/kubernetes/issues/73105#issuecomment-486914764
/test pull-kubernetes-kubemark-e2e-gce-big

**Pull request merged**